tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/AMQPMirrorExpiryQueueTest.java

```diff
@@ -0,0 +1,140 @@
+      return server;
+   }
+
+   @Test
+   public void testExpiryOnMirrorSNF() throws Exception {
+
+      final long numberOfMessages = 10;
+      ActiveMQServer serverA = createServer(5671, "serverA");
+      ActiveMQServer serverB = createServer(6671, "serverB");
+
+      {
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration("A_to_B", "tcp://localhost:6671").setReconnectAttempts(-1).setRetryInterval(10);
+         AMQPMirrorBrokerConnectionElement replica = new AMQPMirrorBrokerConnectionElement().setType(AMQPBrokerConnectionAddressType.MIRROR).setDurable(true);
+         amqpConnection.addElement(replica);
+         serverA.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      {
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration("B_to_A", "tcp://localhost:5671").setReconnectAttempts(-1).setRetryInterval(10);
+         AMQPMirrorBrokerConnectionElement replica = new AMQPMirrorBrokerConnectionElement().setType(AMQPBrokerConnectionAddressType.MIRROR).setDurable(true);
+         amqpConnection.addElement(replica);
+         serverB.getConfiguration().addAMQPConnection(amqpConnection);
+      }
+
+      String queueName = "queue" + RandomUtil.randomString();
+
+      serverA.setIdentity("serverA");
+      serverB.setIdentity("serverB");
+      serverA.start();
+      serverB.start();
+
+      serverA.createQueue(QueueConfiguration.of(queueName).setName(queueName).setRoutingType(RoutingType.ANYCAST));
+      Queue expiryA = serverA.createQueue(QueueConfiguration.of(EXPIRY_QUEUE).setName(EXPIRY_QUEUE).setRoutingType(RoutingType.ANYCAST));
+      Wait.assertTrue(() -> serverB.locateQueue(queueName) != null, 5000, 100);
+      Wait.assertTrue(() -> serverB.locateQueue(EXPIRY_QUEUE) != null, 5000, 100);
+
+      Queue serverAMirrorSNF = serverA.locateQueue("$ACTIVEMQ_ARTEMIS_MIRROR_A_to_B");
+      Queue serverBMirrorSNF = serverB.locateQueue("$ACTIVEMQ_ARTEMIS_MIRROR_B_to_A");
+
+      serverB.stop();
+
+      ConnectionFactory factoryA = CFUtil.createConnectionFactory("CORE", "tcp://localhost:5671");
+      try (Connection connection = factoryA.createConnection()) {
+         connection.setClientID("clientID");
+         Session session = connection.createSession(true, Session.SESSION_TRANSACTED);
+         MessageProducer producer = session.createProducer(session.createQueue(queueName));
+         producer.setTimeToLive(1);
+
+         for (int i = 0; i < numberOfMessages; i++) {
+            producer.send(session.createTextMessage("hello" + i));
+         }
+
+         session.commit();
+      }
+
+      Thread.sleep(10); // waiting a little more than the expiry scan period
+
+      Wait.assertEquals(numberOfMessages, expiryA::getMessageCount, 5000, 100);
+
+      serverB.start();
+
+      Wait.assertEquals(0L, serverAMirrorSNF::getMessageCount, 5000, 100);
+      Wait.assertEquals(0L, serverBMirrorSNF::getMessageCount, 5000, 100);
```

Comment:
This test would still pass supposing the messages did expire in the SnF queue, and were never sent to the mirror, but just didnt get added to the expiry queue. It never verifies the second broker actually gets any of the messages.

I think a far better test would be to use a test peer as substitute for the second broker, not give enough credit for the messages to be sent initially, then verify they remain in the SnF queue whilst the ones on the origin queue expire, and then grant credit and verify they are actually received by the peer.